QuoridorEngine/src/ui/gui/GuiClient.cs

```diff
@@ -0,0 +1,361 @@
+
+            // Update gui wall cells based on new state
+            refreshWallCells(guiFrame.FreeWallCell, false);
+
+            // Update player's wall counter in gui
+            guiFrame.SetPlayerWallCounter(isWhitePlayerTurn, game.GetPlayerWalls(isWhitePlayerTurn));
+        }
+
+        /// <summary>
+        /// Event handler for player pawn movement
+        /// </summary>
+        /// <param name="sender">Player cell to move player pawn</param>
+        /// <param name="e">Event arguments</param>
+        private void MovePlayer(object sender, EventArgs e)
+        {
+            // Ignore if player move is not initiated
+            if (!initPlayerMove) return;
+
+            PlayerCell playerCell = (PlayerCell)sender;
+
+            // Get current player coordinates
+            int currentGameRow = 0, currentGameColumn = 0;
+            if(isWhitePlayerTurn)
+                game.GetWhiteCoordinates(ref currentGameRow, ref currentGameColumn);
+            else
+                game.GetBlackCoordinates(ref currentGameRow, ref currentGameColumn);
+
+            // Construct new quoridor move
+            (int newGameRow, int newGameColumn) = TransformCoordinates.GuiToGamePlayer(playerCell.Row, playerCell.Column);
+            QuoridorMove newMove = new QuoridorMove(currentGameRow, currentGameColumn, newGameRow, newGameColumn, isWhitePlayerTurn);
+
+            try
+            {
+                // Attempt new move execution
+                game.ExecuteMove(newMove);
+            }
+            catch (InvalidMoveException)
+            {
+                // Ignore if invalid
+                return;
+            }
+
+            // Move completed
+            hidePossiblePlayerMoves();
+
+            // Update player pawn location in gui based on last move
+            guiFrame.MovePlayerPawn(isWhitePlayerTurn, playerCell.Row, playerCell.Column);
+
+            // Player's turn has finished
+            switchPlayerTurn();
+        }
+
+        /// <summary>
+        /// Event handler for player pawn click 
+        /// </summary>
+        /// <param name="sender">Clicked player pawn</param>
+        /// <param name="e">Event arguments</param>
+        private void PlayerPawnClicked(object sender, EventArgs e)
+        {
+            bool isWhitePlayer = ((PlayerPawn)sender).IsWhite;
+
+            // Ignore if not player's turn
+            if (isWhitePlayer != isWhitePlayerTurn) return;
+
+            if (initPlayerMove)
+                // Cancel initiated player move
+                hidePossiblePlayerMoves();
+            else
+                // Initiate player move 
+                showPossiblePlayerMoves(isWhitePlayer);
+        }
+
+        /// <summary>
+        /// Update gui board player cells to show player's pawn possible moves
+        /// </summary>
+        /// <param name="isWhitePlayer">Player to show possible moves</param>
+        private void showPossiblePlayerMoves(bool isWhitePlayer)
```

Comment:
This function logically belongs to the class responsible for handling the UI (Board or GUIFrame?) not the handler itself. Move it there and pass the quoridor move list as a parameter